docs/TestingConcepts.md

```diff
@@ -0,0 +1,29 @@
+# Benchmarking Concepts
+
+Benchmark is designed to work easily in different contexts like within and IDE, from the command line, and as part of Github workflows. However, there is more to writing benchmarks than getting a test working and measuring between two points. Is it reproducible? Does it perform similarly at different scales? What about from day to day? Is setup included in the measurement? Is it easy to add to and maintain?
+
+What follows are some concepts that guide the development of Benchmark meant to kept it versatile, simple, and relevant.
+
+### Self-guided API
+The *Bench* API uses the builder pattern to guide the test writer in generating data, executing queries, and fetching results. There is a single API entry point where a user can follow the dots and look at the code-insight and Javadocs that pop up in the IDE. Default properties can be overriden by builder-style "with" methods like *withRowCount()*. A middle ground is taken between text configuration and configuration fully-expressed in code to keep things simple and readable.
+
+### Scale Rather Than Iterations
+Repeating tests can be useful for testing the effects of caching (e.g. load file multiple times; is it faster on subsequent loads?), or overcoming a lack of precision in OS timers (e.g. run a fast function many times and average), or average out variability between runs (there are always anomalies). On the other hand, if the context of the test is processing large data sets, then it's better to measure against large data sets where possible. This provides a benchmark test that's closer to the real thing when it comes to memory consumption, garbage collection, thread usage, and JIT optimizations. Repeating tests, though useful in some scenarios, can have the effect of taking the operation under test out of the benchmark equation because of cached results, resets for each iteration, 
+limited heap usage, or smaller data sets that are too uniform.
+
+### Adjust Scale For Each Test
+When measuring a full set of benchmarks for transforming data, some benchmarks will naturally be faster than others (e.g. sums vs joins). Running all benchmarks at the same scale (e.g. 10 million rows) could yield results where one benchmark takes a minute and another takes 100 milliseconds. Is the 100 ms test meaningful, especially when measured in a JVM? Not really, because there is no time to assess the impact of JVM ergonomics or the effect of OS background tasks. A better way is to set scale multipliers to amplify row count for tests that need it.
```

Comment:
I know from talking to you that "bigger scale is better", but that isn't entirely explicit in this paragraph. (Or maybe I'm over-simplifying and not understanding your concept?)

If I was my normal impatient self, I'd be thinking "tests that return in a shorter amount of timeare better", so may mis-read this section as a tool to scale down and get "quicker" results. It might be better to talk about how you normalize for the rate and that therefore anything less than some threshold of time consumed becomes statistically inaccurate due e.g. to the multiplier on the margin-of-error. And explicitly suggest targeting e.g. 1 min (or whatever) as a good starting point for how long a test "should" take.